app/src/main/java/com/conupods/OutdoorMaps/Services/ActivityComponentBuilder.java

```diff
@@ -0,0 +1,62 @@
+package com.conupods.OutdoorMaps.Services;
+
+import android.app.Activity;
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.widget.LinearLayout;
+import android.widget.RelativeLayout;
+import android.widget.SearchView;
+
+import com.conupods.OutdoorMaps.View.SearchView.AbstractCampusLocationAdapter;
+import com.conupods.R;
+import com.google.android.libraries.places.api.model.Place;
+import com.google.android.libraries.places.widget.Autocomplete;
+import com.google.android.libraries.places.widget.model.AutocompleteActivityMode;
+
+import android.content.Intent;
+import android.widget.SearchView.OnQueryTextListener;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class ActivityComponentBuilder {
+    private SearchView mSearchBar;
+    private LinearLayout mSarchDirectionsOptions;
+
+
+    public SearchView initializeSearchBarWithFocus(SearchView searchBar, Context context, Activity activity, AbstractCampusLocationAdapter abstractCampusLocationAdapter) {
+        mSearchBar = searchBar;
+        mSearchBar.setQueryHint("Where To?");
+        mSearchBar.setFocusable(true);
+        mSearchBar.setIconified(false);
+        mSearchBar.requestFocusFromTouch();
+
+        mSearchBar.setMaxWidth(Integer.MAX_VALUE);
+
+        // listening to search query text change
+        mSearchBar.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                // filter recycler view when query submitted
```

Comment:
Is it necessary to keep this inner class? Inner classes are hard to test and we might need to add more info to it in the future. Although this is your code so I am leaving the judgement to you, if you think it will remain untouched, I wouldn't change it!

Reply:
would making a lambda make this easier to test?